columnphysics/icepack_therm_itd.F90

```diff
@@ -1020,59 +1019,48 @@ subroutine lateral_melt (dt,         ncat,       &
          d_afsd_latm(:) = c0
       end if
 
-      if (tr_fsd .and. fside < c0) then
+      if (tr_fsd .and. wlat > puny) then
          flag = .true.
-
-
-! echmod - using category values would be preferable to the average value
-         ! Compute average enthalpy of ice (taken from add_new_ice)
-         if (sss > c2 * dSin0_frazil) then
-            Si0 = sss - dSin0_frazil
-         else
-            Si0 = sss**2 / (c4*dSin0_frazil)
-         endif
-         Ti = min(liquidus_temperature_mush(Si0/phi_init), -p1)
-         qi0 = enthalpy_mush(Ti, Si0)
-
+         ! for FSD rside and fside not yet computed correctly, redo here
+         fside = c0
          do n = 1, ncat
-            if (ktherm == 2) then  ! mushy
-               do k = 1, nilyr
-                  !qin(n) = qin(n) &
-                  !       + trcrn(nt_qice+k-1,n)*vicen(n)/real(nilyr,kind=dbl_kind)
-                  qin(n) = qi0
-               enddo
-            else
-               qin(n) = -rhoi*Lfresh
-            endif
 
-            if (qin(n) < -puny) G_radialn(n) = -fside/qin(n) ! negative
+            G_radialn(n) = -wlat ! negative
 
-            if (G_radialn(n) < -puny) then
+            if (any(afsdn(:,n) < c0)) print*,'lateral_melt B afsd < 0',n
 
+            bin1_arealoss = -trcrn(nt_fsd+1-1,n) * aicen(n) * dt &
+                             * G_radialn(n) / floe_binwidth(1)
 
-               if (any(afsdn(:,n) < c0)) print*,&
-                 'lateral_melt B afsd < 0',n
+            delta_an(n) = c0
+            do k = 1, nfsd
+               delta_an(n) = delta_an(n) + ((c2/floe_rad_c(k))*aicen(n) &
+                    * trcrn(nt_fsd+k-1,n)*G_radialn(n)*dt) ! delta_an < 0
+            end do
 
-               cat1_arealoss = -trcrn(nt_fsd+1-1,n) * aicen(n) * dt &
-                             * G_radialn(n) / floe_binwidth(1)
+            ! add negative area loss from fsd
+            delta_an(n) = delta_an(n) - bin1_arealoss
 
-               delta_an(n) = c0
-               do k = 1, nfsd
-                  delta_an(n) = delta_an(n) + ((c2/floe_rad_c(k))*aicen(n) &
-                                            * trcrn(nt_fsd+k-1,n)*G_radialn(n)*dt) ! delta_an < 0
-               end do
+            if (delta_an(n) > c0) print*,'ERROR delta_an > 0', delta_an(n)
```

Comment:
Also needs to use warning system.  Please check for other print * implementations.  I think a few exist already and should have been caught earlier.  A grep of the current icepack suggests they are in fsd and therm_itd.  Would you mind fixing them with this PR?  Also, please make sure any new calls to subroutines within Icepack check the abort flag and return if true.  Thanks.